src/Extensions/SiteTreeSubsites.php

```diff
@@ -53,6 +53,11 @@ class SiteTreeSubsites extends DataExtension
     private static $many_many_extraFields = [
         'CrossSubsiteLinkTracking' => ['FieldName' => 'Varchar']
     ];
+    
+    /**
+    * Used to cache the result of a heavily called database query
+    */
+    private bool $subsitesExist;
```

Comment:
Note that because extensions are used as singletons, and subsites existing is a global thing, it *is* appropriate to cache this as a member property as this means all pages this is being called on will get the benefit of the cache.